lib/ransack_ui/ransack_overrides/helpers/form_builder.rb

```diff
@@ -198,6 +198,12 @@ def attribute_collection_for_base(base, attributes=nil)
       rescue UntraversableAssociationError => e
         nil
       end
+
+      private
+
+      def autocomplete_source
+        @autocomplete_source ||= @foreign_klass && @foreign_klass._ransack_autocompletes_through
```

Comment:
I'm not sure if the instance variable for `@foreign_klass` is going to be a safe approach.

The method `def attribute_collection_for_base(base, attributes=nil)` takes a base object: if this is different between method calls and we use it to redefine this then things could get mixed up?